File: benchmark/c/task7_error.c
#include <stdlib.h>
#include <errno.h>

typedef struct {
    int success;
    int value;
    const char* error;
} ParseResult;

ParseResult parse_int(const char* s) {
    char* end;
    errno = 0;
    long val = strtol(s, &end, 10);

    if (errno != 0 || *end != '\0')
        return (ParseResult){0, 0, "Invalid integer format"};

    return (ParseResult){1, (int)val, NULL};
}

File: benchmark/matlab/task7_error.m
function [success, value, error] = parse_int(s)
    value = str2double(s);
    if isnan(value) || floor(value) ~= value
        success = false;
        value = 0;
        error = 'Invalid integer format';
    else
        success = true;
        value = int32(value);
        error = '';
    end
end
